sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java

```diff
@@ -256,40 +252,28 @@ private void initStorageAccountIdentity(IdentityType identityType) {
             throw logger.logExceptionAsError(new IllegalArgumentException("Invalid argument: " + identityType));
         }
         if (isInCreateMode()) {
-            if (createParameters.identity() == null
-                || createParameters.identity().type() == null
-                || createParameters.identity().type().equals(IdentityType.NONE)
-                || createParameters.identity().type().equals(identityType)) {
-                createParameters.withIdentity(new Identity().withType(identityType));
+            if (storageAccount.createParameters.identity() == null
+                || storageAccount.createParameters.identity().type() == null
+                || storageAccount.createParameters.identity().type().equals(IdentityType.NONE)
+                || storageAccount.createParameters.identity().type().equals(identityType)) {
+                storageAccount.createParameters.withIdentity(new Identity().withType(identityType));
             } else {
-                createParameters.withIdentity(new Identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED));
+                storageAccount.createParameters.withIdentity(new Identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED));
             }
         } else {
-            if (updateParameters.identity() == null
-                || updateParameters.identity().type() == null
-                || updateParameters.identity().type().equals(IdentityType.NONE)
-                || updateParameters.identity().type().equals(identityType)) {
-                Identity identity = Objects.isNull(updateParameters.identity()) ? new Identity().withType(identityType) : updateParameters.identity().withType(identityType);
-                updateParameters.withIdentity(identity);
+            if (storageAccount.updateParameters.identity() == null
+                || storageAccount.updateParameters.identity().type() == null
+                || storageAccount.updateParameters.identity().type().equals(IdentityType.NONE)
+                || storageAccount.updateParameters.identity().type().equals(identityType)) {
+                Identity identity = Objects.isNull(storageAccount.updateParameters.identity()) ? new Identity().withType(identityType) : storageAccount.updateParameters.identity().withType(identityType);
+                storageAccount.updateParameters.withIdentity(identity);
             } else {
-                updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+                storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
             }
         }
     }
 
     private Boolean isInCreateMode() {
         return Objects.isNull(storageAccount.innerModel().id());
     }
```

Comment:
`storageAccount.isInCreateMode()`?

Reply:
The method `isInCreateMode()` removed from `StorageAccountMsiHandler`.